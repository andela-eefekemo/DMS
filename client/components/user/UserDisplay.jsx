import React from 'react';
import { Modal } from 'react-materialize';
import PropTypes from 'prop-types';

import InputField from '../common/InputField';

const UserDisplay = (
  { onSubmit, onChange, firstName, lastName, email }) => {
  return (
    <div>
      <div className="user-card card">
        <h5 className="center"> User Profile </h5>
        <div className="row profile">
          <div className="col l8 m8 s12">
            <InputField
              name="firstName"
              value={firstName}
              placeholder="First Name"
              className="validate form-design"
              type="text" onChange={onChange} />
          </div>
          <div className="col l8 m8 s12">
            <InputField
              name="lastName"
              value={lastName}
              placeholder="Last Name"
              className="validate form-design"
              type="text" onChange={onChange} />
          </div>
          <div className="col l8 m8 s12">
            <InputField
              name="email"
              value={email}
              placeholder="Email"
              className="validate form-design"
              type="email" onChange={onChange} />
          </div>
        </div>
        <div className="row user-action">
          <div className="col l6 m6 s12 input-field center">
            <Modal
              trigger={
                <a
                  className="waves-effect btn button-design"
                  data-target="passwordModal">
                  Change Password
                  </a>
              }
            >
              <div>
                <h5 className="center"> Set Password</h5>
                <div className="row">
                  <div className="col l12 m12 s12">
                    <InputField
                      name="oldPassword"
                      placeholder="Old Password"
                      className="validate form-design"
                      type="password" onChange={onChange} />
                    <InputField
                      name="password"
                      placeholder="Password"
                      className="validate form-design"
                      type="password" onChange={onChange} />
                  </div>
                  <div className="col l12 m12 s12">
                    <InputField
                      name="confirmPassword"
                      placeholder="Re-Type Password"
                      className="validate form-design"
                      type="password" onChange={onChange} />
                  </div>
                </div>
                <div className="center">
                  <button
                    className="waves-effect btn button-design"
                    type="submit" onClick={onSubmit}>
                    Save
                  </button>
                </div>
              </div>
            </Modal>
          </div>
          <div className="col l6 m6 s12 input-field center">
            <button
              className="waves-effect btn button-design"
              type="submit" onClick={onSubmit} >
              Save
              </button>
          </div>
        </div>
      </div>
    </div >
  );
};

UserDisplay.propTypes = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  email: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default UserDisplay;
