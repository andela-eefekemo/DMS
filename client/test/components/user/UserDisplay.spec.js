/* global expect jest test */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserDisplay from '../../../components/user/UserDisplay';

jest.dontMock('../../../components/user/UserDisplay');

describe('UserDisplay Component', () => {
  test('should match the user Display snapshot', () => {
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const component = shallow(
      <UserDisplay
        firstName={'Efe'}
        lastName={'Eguono'}
        email={'email'}
        onChange={onChange}
        onSubmit={onSubmit} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
