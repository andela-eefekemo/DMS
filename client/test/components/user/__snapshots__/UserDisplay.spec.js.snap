// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay Component should match the user Display snapshot 1`] = `
<div>
  <div
    className="user-card card"
  >
    <h5
      className="center"
    >
       User Profile 
    </h5>
    <div
      className="row profile"
    >
      <div
        className="col l8 m8 s12"
      >
        <InputField
          className="validate form-design"
          name="firstName"
          onChange={[Function]}
          placeholder="First Name"
          type="text"
          value="Efe"
        />
      </div>
      <div
        className="col l8 m8 s12"
      >
        <InputField
          className="validate form-design"
          name="lastName"
          onChange={[Function]}
          placeholder="Last Name"
          type="text"
          value="Eguono"
        />
      </div>
      <div
        className="col l8 m8 s12"
      >
        <InputField
          className="validate form-design"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value="email"
        />
      </div>
    </div>
    <div
      className="row user-action"
    >
      <div
        className="col l6 m6 s12 input-field center"
      >
        <Modal
          actions={
            Array [
              <Button
                flat={true}
                modal="close"
                node="button"
                waves="light"
            >
                Close
            </Button>,
            ]
          }
          bottomSheet={false}
          fixedFooter={false}
          modalOptions={Object {}}
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal"
              id="update-password"
            >
              Change Password
            </a>
          }
        >
          <div>
            <h5
              className="center"
            >
               Set Password
            </h5>
            <div
              className="row"
            >
              <div
                className="col l12 m12 s12"
              >
                <InputField
                  className="validate form-design"
                  name="oldPassword"
                  onChange={[Function]}
                  placeholder="Old Password"
                  type="password"
                />
                <InputField
                  className="validate form-design"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </div>
              <div
                className="col l12 m12 s12"
              >
                <InputField
                  className="validate form-design"
                  name="confirmPassword"
                  onChange={[Function]}
                  placeholder="Re-Type Password"
                  type="password"
                />
              </div>
            </div>
            <div
              className="center"
            >
              <button
                className="btn button-design modal-action modal-close"
                id="password-button"
                onClick={[Function]}
                type="submit"
              >
                Save
              </button>
            </div>
          </div>
        </Modal>
      </div>
      <div
        className="col l6 m6 s12 input-field center"
      >
        <button
          className="waves-effect btn button-design"
          id="update-button"
          onClick={[Function]}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
